src/models/graph.ts

```diff
@@ -118,6 +120,28 @@ export class Graph<N extends INodeBase, E extends IEdgeBase> implements IGraph<N
     return this._edges.getOne(id);
   }
 
+  /**
+   * Returns a list of selected nodes.
+   *
+   * @return {INode[]} List of selected nodes
+   */
+  getSelectedNodes(): INode<N, E>[] {
+    const nodes = this.getNodes();
+    const selectedNodes = nodes.filter((node) => node.isSelected());
+    return selectedNodes;
```

Comment:
I think it would be valuable to use the inner model for filtering. What I mean by this is to change this to

```typescript
return this.getNodes((node) => node.isSelected());
```

The inner model (`nodes`) keeps the ordering + it might have some other optimizations in the future. 

And what I commented in the main thread, it might be useful to have `getHoveredNodes` and `getHoveredEdges` - for all of these you can use the `getNodes` and `getEdges` with custom filter.